src/Core/src/Platform/iOS/MauiScrollView.cs

```diff
@@ -1,33 +1,169 @@
+		
+		/// <summary>
+		/// The width constraint used in the last arrange operation.
+		/// Used to determine if a re-arrange is needed when the frame changes.
+		/// </summary>
 		double _lastArrangeWidth;
 
+		/// <summary>
+		/// The current safe area padding values derived from iOS's AdjustedContentInset.
+		/// Represents the areas that should be avoided when placing content (e.g., status bar, home indicator).
+		/// </summary>
+		SafeAreaPadding _safeArea = SafeAreaPadding.Empty;
+		
+		/// <summary>
+		/// Flag indicating whether the safe area needs to be recalculated.
+		/// Set to true when iOS notifies us of safe area changes.
+		/// </summary>
+		bool _safeAreaInvalidated = true;
+		
+		/// <summary>
+		/// Flag indicating whether this scroll view should apply safe area adjustments to its content.
+		/// Only true when not nested in another scroll view and safe area is not empty.
+		/// </summary>
+		bool _appliesSafeAreaAdjustments;
+		
+		/// <summary>
+		/// The previous effective user interface layout direction (LTR/RTL).
+		/// Used to detect when the layout direction changes and trigger appropriate content repositioning.
+		/// </summary>
 		UIUserInterfaceLayoutDirection? _previousEffectiveUserInterfaceLayoutDirection;
 
+		/// <summary>
+		/// Weak reference to the cross-platform layout that manages the content of this scroll view.
+		/// Weak reference prevents circular references and allows proper garbage collection.
+		/// </summary>
 		WeakReference<ICrossPlatformLayout>? _crossPlatformLayoutReference;
 
+		/// <summary>
+		/// Gets or sets the cross-platform layout that manages the content of this scroll view.
+		/// The layout is responsible for measuring and arranging the scroll view's content.
+		/// </summary>
 		ICrossPlatformLayout? ICrossPlatformLayoutBacking.CrossPlatformLayout
 		{
 			get => _crossPlatformLayoutReference != null && _crossPlatformLayoutReference.TryGetTarget(out var v) ? v : null;
 			set => _crossPlatformLayoutReference = value == null ? null : new WeakReference<ICrossPlatformLayout>(value);
 		}
 
+		/// <summary>
+		/// Internal accessor for the cross-platform layout. Provides typed access to the layout manager.
+		/// </summary>
 		internal ICrossPlatformLayout? CrossPlatformLayout => ((ICrossPlatformLayoutBacking)this).CrossPlatformLayout;
 
+		/// <summary>
+		/// Initializes a new instance of the MauiScrollView class.
+		/// </summary>
+		public MauiScrollView()
+		{
+		}
+
+		/// <summary>
+		/// Determines whether this scroll view should respond to safe area changes.
+		/// Returns false if this scroll view is nested within another scroll view,
+		/// as nested scroll views should not apply their own safe area adjustments.
+		/// </summary>
+		/// <returns>True if this scroll view should apply safe area adjustments, false otherwise.</returns>
+		bool RespondsToSafeArea()
+		{
+			return !(_scrollViewDescendant ??= Superview.GetParentOfType<UIScrollView>() is not null);
```

Comment:
In the MaiuView, we got code like this:

```
return (_respondsToSafeArea = RespondsToSelector(new Selector("safeAreaInsets"))).Value;
```

Should this also check to see if the OS supports it, or is this not needed since this was iOS 11+. Should we remove from maui view and then drop that whole check?